draft-normington-jsonpath-latest.xml

```diff
@@ -218,6 +225,94 @@ dot-child-name = TBD
+                  <name>Syntax</name>
+                  <t>A union matcher consists of one or more union elements.
+                  </t>
+                  <sourcecode type="abnf">
+matcher =/ union
+union = %x5B ws union-elements ws %x5D ; [...]
+ws = *%x20                             ; zero or more spaces
+union-elements = union-element /
+                 union-element ws %x2C ws union-elements
+                                       ; ,-separated list
+                  </sourcecode>
+               </section>       
+               <section numbered="true" toc="default">
+                  <name>Semantics</name>
+                  <t>A union matches any node which is matched by at least one of the union elements and selects the concatenation of the
+                     lists (in the order of the matchers) of descendants selected by the union elements.
+                  </t>
+               </section>
+               <section numbered="true" toc="default" anchor="unionchild">
+                  <name>Union Child</name>
+                  <section numbered="false" toc="exclude">
+                     <name>Syntax</name>
+                     <t>A union child is a union element consisting of a quoted string.
+                     </t>
+                     <sourcecode type="abnf">
+union-element = union-child ; see below for more alternatives
+union-child = %x27 *single-quoted %x27 / ; 'string'
+              %x22 *double-quoted %x22   ; "string"
+
+single-quoted = %x5C %x27 / ; \' (escaped single quote)
+                %x5C %x5C / ; \\ (escaped reverse solidus)
+                %x00-26 / %x28-5B %x5D-D7FF / %xE000-10FFFF
+                            ; any Unicode code point except ' or \
+
+double-quoted = %x5C 5x22 / ; \" (escaped double quote)
+                %x5C 5x5C / ; \\ (escaped reverse solidus)
+                %x00-21 / %x23-5B / %x5D-D7FF / %xE000-10FFFF
+                            ; any Unicode code point except " or \
+                     </sourcecode>
+                     <t>Note: the syntax does not allow any character other than a quote or reverse solidus to be escaped.
+                     </t>
+                  </section>       
+                  <section numbered="false" toc="exclude">
+                     <name>Semantics</name>
+                     <t>If the union child is a quoted string, the string must first have the surrounding quotes removed and then be unescaped
+                        as describe below.
+                        The union child matches any node which is an object having a key equal to the unescaped value of the union child string
+                        and selects the value corresponding to that key.
+                        It does not match a node which is not an object.
+                     </t>
+                     <t>After the surrounding quotes have been removed, a string is unescaped by replacing each escaped quote by the corresponding
```

Comment:
Maybe it's better to fully describe the unescaping process at the beginning of the section without splitting it with matching description? So that at first we describe how do we _get_ the key and then - how we _use_ the key.

Reply:
Good idea.